solarforecastarbiter/metrics/tests/test_preprocessing.py

```diff
@@ -1653,3 +1726,268 @@ def test_apply_fill_unstratified_dataframe(data, method, exp, exp_count,
                                              data.index[0], data.index[-1])
     pd.testing.assert_frame_equal(result, exp)
     assert count == exp_count
+
+
+OUTAGE_FORECAST = datamodel.Forecast(
+   site=_site_metadata(), name='dummy fx', variable='ghi',
+   interval_value_type='instantaneous',
+   interval_length=pd.Timedelta('1h'),
+   interval_label='beginning',
+   issue_time_of_day=dt.time(hour=5),
+   lead_time_to_start=pd.Timedelta('1h'),
+   run_length=pd.Timedelta('12h')
+)
+
+
+@pytest.mark.parametrize("forecast,start,end,expected", [
+    (OUTAGE_FORECAST,
+     pd.Timestamp('2021-01-01T00:00Z'),
+     pd.Timestamp('2021-01-02T00:00Z'),
+     pd.date_range(
+        '2020-12-31T17:00Z',
+        '2021-01-01T17:00Z',
```

Comment:
1 too many issue times in all of these tests. Could add closed='left' to the `date_range` call in `forecast_report_issue_times`. I'd leave the tests with closed=None (default) and change the times.

Reply:
Thanks! There are issues with a few of them and I had to adjust the `date_range` call to use `closed=left` as well as end at `report.end - forecast.lead_time_to_start`. The forecast above has a 1 hour lead time and 12 hr run length, so 2021-01-01T17:00Z is still within the report range for the first 6 hours of data.